### ISSUE TITLE
Payroll - issues

### ISSUE THREAD

**Issue by ronanq** [XeroAPI/Xero-Net#73]
As reported by a user:

* In Xero-Net for, the end point for payslips is payroll.xro/1.0/Payslips
* A payslip will not Deserialize to the Payslip object after returning a successful JSON request

**Comment by nichbeau** [comment]
The issue here is that the wrapper is always expecting Payslips but Xero always refrences Payslip (i.e. no s (plural)).

To get around this we had to do the following:
-In Endpoints -> PayslipsEndpoint.cs change end point to be like below (with no S). 
        public PayslipsEndpoint(XeroHttpClient client)
            : base(client, "payroll.xro/1.0/Payslip")
-To deserialize we just ended up hacking this a bit to save time, so maybe when we have time or if someone else wants to, we could change the structure to accept Payslip. Anyway for now in DeserializeTypeRefJson.cs, function StringToType when retrieving the propertyName if it is Payslip we simply change it to Payslips. 

 var strTypeLength = strType.Length;
            while (index < strTypeLength)
            {
                var propertyName = JsonTypeSerializer.ParseJsonString(strType, ref index);

                if (propertyName.Equals("Payslip"))
                {
                    propertyName = "Payslips";
                }
